f5_openstack_agent/lbaasv2/drivers/bigip/service_adapter.py

```diff
@@ -286,6 +286,13 @@ def _map_pool(self, loadbalancer, lbaas_pool, lbaas_hm):
             pool["loadBalancingMode"] = self._get_lb_method(
                 lbaas_pool["lb_algorithm"])
 
+        # If source_ip lb method, add SOURCE_IP persistence to ensure
+        # source IP loadbalancing. See issue #344 for details.
+        if lbaas_pool["lb_algorithm"].upper() == 'SOURCE_IP':
```

Comment:
Per line 285, "lb_algorithm" may not be present in the dictionary.  Convert line 291 to protect against a KeyError exception.  Perhaps lbaas_pool.get("lb_algorithm", "") == 'SOURCE_IP'.